golem_messages/validators.py

```diff
@@ -19,3 +19,155 @@ def validate_integer(field_name, value):
+    assert value_type in [int, float]
+    return isinstance(value, value_type) and value > 0
+
+
+def validate_positive_integer_value(field_name, value):
+    if not (is_number_positive(value, int) or value == 0):
+        raise exceptions.FieldError(
+            "should be a positive integer field",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_not_negative_integer_value(field_name, value):
+    if not (is_number_positive(value, int) or value == 0):
+        raise exceptions.FieldError(
+            "should be a not negative integer field",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_not_negative_number(field_name, value):
+    if float(value) < 0:
+        raise exceptions.FieldError(
+            "Should be a not negative number",
+            field_name=field_name,
+            value=value
+        )
+
+
+def validate_boolean(field_name, value):
+    if not isinstance(value, bool):
+        raise exceptions.FieldError(
+            f"Should be boolean",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_list(field_name, value, min_length, max_length):
+    if not (isinstance(value, list) and
+            (min_length <= len(value) <= max_length)):
+        raise exceptions.FieldError(
+            f"Should be a list with minimum {min_length} "
+            f"element and maximum {max_length} elements.",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_list_of_positive_integers(
+        field_name,
+        value,
+        min_length=1,
+        max_length=float('inf')
+):
+    validate_list(field_name, value, min_length, max_length)
+    for element in value:
+        if not is_number_positive(element, int):
+            raise exceptions.FieldError(
+                f"Should be a list with positive integers",
+                field=field_name,
+                value=value
+            )
+
+
+def validate_list_of_non_negative_floats(
+        field_name,
+        value,
+        min_length=1,
+        max_length=float('inf')
+):
+    validate_list(field_name, value, min_length, max_length)
+    for element in value:
+        if not (is_number_positive(element, float) or float(element) == 0.0):
+            raise exceptions.FieldError(
+                f"Should be a list of floats equal or greater 0.0",
+                field=field_name,
+                value=element
+            )
+
+
+def validate_list_of_non_negative_numbers(
+        field_name,
+        value,
+        min_length=1,
+        max_length=float('inf')
+):
+    validate_list(field_name, value, min_length, max_length)
+    for element in value:
+        if float(element) < 0.0:
+            raise exceptions.FieldError(
+                f"Should be a list of numbers equal or greater 0.0",
+                field=field_name,
+                value=element
+            )
+
+
+def validate_enum(field_name, value, enum_class):
```

Comment:
This somehow duplicates `Message.deserialize_slot()`:
https://github.com/golemfactory/golem-messages/blob/fb4afbc8e17265895adfb3059c67eb91c4062d70/golem_messages/message/base.py#L368-L378

Is it possible to combine those two approaches (reduce duplication)?

Reply:
Hmm, it might be hard because `BlenderScriptPackage` doesn't inherit from `Message()` class where this method is implemented so I can't reuse `deserialzie_slot` in this dict.